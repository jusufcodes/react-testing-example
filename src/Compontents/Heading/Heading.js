import React from "react";

export const Heading = (props) => {
  const level = props.level ? props.level : 1;

  const Elem = `h${level}`;

  return <Elem>{props.children}</Elem>
};

export const Avatar = (props) => {
  return props.src ? <img src={props.src} alt={props.alt} /> : null;
};