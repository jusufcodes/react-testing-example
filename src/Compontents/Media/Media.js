import React from "react";

export const Media = (props) => {
  return props.src ? <img src={props.src} alt={props.alt ? props.alt : ""} /> : null;
};
