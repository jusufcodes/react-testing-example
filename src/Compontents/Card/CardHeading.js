import React from "react";
import {Media} from "../Media/Media";
import {Heading} from "../Heading/Heading";

export const CardHeading = (props) => {
  return (props.avatar || props.heading || props.subHeading) ? (
    <div className={"cardHeading"}>
      {props.avatar && <Media src={props.avatar} />}
      {props.heading && <Heading level={2}>{props.heading}</Heading>}
      {props.subHeading && <Heading level={3}>{props.subHeading}</Heading>}
    </div>
  ) : null;
};