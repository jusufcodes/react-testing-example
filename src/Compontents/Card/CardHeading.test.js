import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import {CardHeading} from "./CardHeading";
import {Heading} from "../Heading/Heading";
import {Media} from "../Media/Media";

describe("CardHeading Component", () => {

  test("without content", () => {
    const heading = <CardHeading/>;

    expect(shallow(heading).isEmptyRender()).toEqual(true);
  });

  test("whith heading", () => {
    const heading = <CardHeading
      heading={"Card Example"}
    />;

    const shallowHeading = shallow(heading);

    expect(shallowHeading.children()).toHaveLength(1);
    expect(shallowHeading.find(Heading).prop("level")).toEqual(2);

    const snapshot = renderer.create(heading).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test("with sub heading", () => {
    const heading = <CardHeading
      subHeading={"Demonstration of TDD"}
    />;

    const shallowHeading = shallow(heading);

    expect(shallowHeading.children()).toHaveLength(1);
    expect(shallowHeading.find(Heading).prop("level")).toEqual(3);

    const snapshot = renderer.create(heading).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test("with image", () => {
    const heading = <CardHeading
      avatar={"https://via.placeholder.com/60x60"}
    />;

    const shallowHeading = shallow(heading);

    expect(shallowHeading.children()).toHaveLength(1);
    expect(shallowHeading.find(Media)).toHaveLength(1);

    const snapshot = renderer.create(heading).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test("with all props", () => {
    const heading = <CardHeading
      avatar={"https://via.placeholder.com/60x60"}
      heading={"Card Example"}
      subHeading={"Demonstration of TDD"}
    />;

    const shallowHeading = shallow(heading);

    expect(shallowHeading.children()).toHaveLength(3);
    expect(shallowHeading.find(Heading)).toHaveLength(2);
    expect(shallowHeading.find(Media)).toHaveLength(1);

    const snapshot = renderer.create(heading).toJSON();
    expect(snapshot).toMatchSnapshot();
  })
});