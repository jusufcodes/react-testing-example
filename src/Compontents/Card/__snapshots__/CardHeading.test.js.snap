// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeading Component whith heading 1`] = `
<div
  className="cardHeading"
>
  <h2>
    Card Example
  </h2>
</div>
`;

exports[`CardHeading Component with all props 1`] = `
<div
  className="cardHeading"
>
  <img
    alt=""
    src="https://via.placeholder.com/60x60"
  />
  <h2>
    Card Example
  </h2>
  <h3>
    Demonstration of TDD
  </h3>
</div>
`;

exports[`CardHeading Component with image 1`] = `
<div
  className="cardHeading"
>
  <img
    alt=""
    src="https://via.placeholder.com/60x60"
  />
</div>
`;

exports[`CardHeading Component with sub heading 1`] = `
<div
  className="cardHeading"
>
  <h3>
    Demonstration of TDD
  </h3>
</div>
`;
